src/linux_service.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Set of utilities related to managing Linux services."""
+
+import logging
+import os
+from subprocess import CalledProcessError
+from typing import Optional
+
+from jinja2 import Template
+
+from utils import shell
+
+SERVICE_TEMPLATE = "./templates/service.j2"
+
+logger = logging.getLogger(__name__)
+
+
+class Service:
+    """Class for representing a Linux Service."""
+
+    def __init__(self, name: str):
+        """Sets service name."""
+        self.name = name
+
+    def is_active(self) -> bool:
+        """Returns whether service is active."""
+        try:
+            response = shell(f"systemctl is-active {self.name}")
+            return response == "active\n"
+        except CalledProcessError:
+            return False
+
+    def create(
+        self, command: str, user: str, description: str, exec_stop_post: Optional[str] = None
+    ) -> None:
+        """Creates a linux service."""
+        with open(SERVICE_TEMPLATE, "r") as template:
+            service_content = Template(template.read()).render(
+                command=command, user=user, description=description, exec_stop_post=exec_stop_post
+            )
+        with open(f"/etc/systemd/system/{self.name}.service", "w") as service:
+            service.write(service_content)
+        _systemctl_daemon_reload()
+
+    def delete(self) -> None:
+        """Deletes a linux service."""
+        os.remove(f"/etc/systemd/system/{self.name}.service")
+
+    def restart(self) -> None:
+        """Restarts a linux service."""
+        _systemctl("restart", self.name)
+        logger.info("Service %s restarted", self.name)
+
+    def stop(self) -> None:
+        """Stops a linux service."""
+        _systemctl("stop", self.name)
+        logger.info("Service %s stopped", self.name)
+
+    def enable(self) -> None:
+        """Enables a linux service."""
+        _systemctl("enable", self.name)
+        logger.info("Service %s enabled", self.name)
+
+
+def _systemctl(action: str, service_name: str) -> None:
```

Comment:
As mentioned in another comment, this could return the output and it would be a bit more useful.

Reply:
Fixed